spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Trigger.scala

```diff
@@ -99,5 +99,29 @@ object Trigger {
       }
     }
   }
+
+  /**
+   * A trigger that triggers an action when validation score larger than "max" score
+   * @param max max score
+   */
+  def maxScore(max: Float): Trigger = {
+    new Trigger() {
+      override def apply(state: Table): Boolean = {
+        state[Float]("score") > max
+      }
+    }
+  }
+
+  /**
+   * A trigger that triggers an action when training loss less than "min" loss
+   * @param min min loss
+   */
+  def minLoss(min: Float): Trigger = {
+    new Trigger() {
+      override def apply(state: Table): Boolean = {
+        state[Float]("Loss") < min
+      }
+    }
+  }
```

Comment:
May not be addressed in this pr, but we could give it a thought of below cases: 
- we are at some point, where the loss is below "min" and will actually continue to drop if we train longer
- we are at some point, where the loss is just a little bit above the "min" but never goes below "min". It trains forever.